File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  const isQuestion = message.trim().endsWith("?");
  const isSilence = message.trim().length == 0;
  const isYelling = /[A-Z]+/.test(message) && /^[^a-z]*$/.test(message);

  if (isSilence) return "Fine. Be that way!";
  if (isQuestion && isYelling) return "Calm down, I know what I'm doing!";
  if (isQuestion) return "Sure.";
  if (isYelling) return "Whoa, chill out!";

  return "Whatever.";
}

File: solutions/elixir/bob/1/lib/bob.ex
defmodule Bob do
  @spec hey(String.t()) :: String.t()
  def hey(input) do
    str = String.trim(input)
    cond do 
      str == "" -> "Fine. Be that way!"
      upcase?(str) -> cond do
        question?(str) -> "Calm down, I know what I'm doing!"
        true -> "Whoa, chill out!"
      end 
      question?(str) -> "Sure."
      true -> "Whatever."
    end
  end

  defp text?(str), do: str =~ ~r([a-zA-ZА-Яа-яЁё]) 
  defp upcase?(str), do: text?(str) && str == String.upcase(str) 
  defp question?(str), do: String.ends_with?(str, "?")
end
